lib/dataContract/DataContractFactory.js

```diff
@@ -59,7 +61,18 @@ class DataContractFactory {
    * @return {Promise<DataContract>}
    */
   async createFromSerialized(payload, options = { }) {
-    const rawDataContract = decode(payload);
+    let rawDataContract;
+    try {
+      rawDataContract = decode(payload);
+    } catch (error) {
+      throw new InvalidDataContractError([
+        new SerializedObjectParsingError(
+          SerializedObjectParsingError.OBJECT_TYPE.DATA_CONTRACT,
```

Comment:
I think we don't need it anymore since we know the context with`InvalidDataContractError`?